new-log-viewer/src/services/LogFileManager.ts

```diff
@@ -49,13 +55,17 @@ const loadFile = async (fileSrc: FileSrcType)
  * Class to manage the retrieval and decoding of a given log file.
  */
 class LogFileManager {
+    readonly #fileName: string;
+
+    readonly #numEvents: number = 0;
+
     readonly #pageSize: number;
 
-    readonly #fileName: string;
+    #queryId: number = 0;
```

Comment:
Shall we leave some docs somewhere (maybe some at the `submitQuery` method) to explain why we need this field? e.g.

> Due to JavaScript's single-threaded nature, there is no way to receive new messages or to intercept the handling of any `onmessage` calls until the callback returns.
...
> To allow the service worker to respond to users' cancellations or new queries in a timely manner, here we assign each query an ID before it starts.
...
> The ID is incremented every time a new query is received from the user.
...
> Before a search task perform its work, it should check whether its query ID matches the latest ID in the LogFileManager... If there has been a new query, the search task shall abort immediately without performing any searches and scheduling for the next chunk.